include/frg/vector.hpp

```diff
@@ -137,6 +158,12 @@ T &vector<T, Allocator>::push(T &&element) {
 	return *pointer;
 }
 
+template<typename T, typename Allocator>
+template<typename... Args>
+T &vector<T, Allocator>::emplace_back(Args &&... args) {
+        return push(T(std::forward<Args>(args)...));
```

Comment:
This does not really emplace: it fails for non-moveable types. Use placement new to construct the T object in-place.